spec/java/util/LinkedList.lsl

```diff
@@ -0,0 +1,666 @@
+			result = action LIST_GET(storage, 0);
+		}
+	}
+	
+	
+	fun element (): Object {
+		result = getFirst();	
+	}
+	
+	
+	fun poll(): Object {
+		if(size==0)
+		{
+			result = null;
+		}
+		else 
+		{
+			result = unlinkAny(0);
+		}
+	}
+	
+	
+	fun remove(): Object {
+		result = removeFirst();
+	}
+	
+	
+	fun offer(e: Object): boolean {
+	 	add(e);
+		result = true;
+	}
+
+	
+	fun offerFirst(e: Object): boolean {
+		addFirst(e);
+		result = true;
+	}
+
+
+	fun offerLast(e: Object): boolean {
+		addLast(e);
+		result = true;
+	}
+	
+	
+	fun peekFirst(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = action LIST_GET(storage, 0);
+		}
+	}
+	
+	
+	fun peekLast(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = action LIST_GET(storage, size-1);
+		}
+	}
+
+
+	fun pollFirst(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+			result = unlinkAny(0);
+		}
+	}
+	
+	
+	fun pollLast(): Object {
+		if(size == 0)
+		{
+			result = null;
+		}
+		else
+		{
+ 			result = unlinkAny(size-1);
+		}
+	}
+	
+	
+	fun push(e: Object): void {
+		addFirst(e);
+	}
+
+
+	fun pop(): Object {
+		result = removeFirst();
+	}
+	
+	
+	fun removeFirstOccurrence(o: Object): boolean {
+		result = remove(o);
+	}
+	
+	
+	fun removeLastOccurrence(o: Object): boolean {
+		//I need think about this
+	}
+	
+	
+	//We need add type: typealias ArrayObject = array<Object>;
+	fun toArray(a: ArrayObject): ArrayObject {
+		result = action LIST_TO_ARRAY(storage, a);
+	}
+
+	
+	fun spliterator(): Spliterator {
+		result = new LLSpliterator(state=Initialized,
+		//This is right ? "parent=self"
+		parent=self,
+		est=-1,
+		expectedModCount=0);
+	}
+
+
+	fun listIterator(index: int): ListIterator {
+		checkPositionIndex(index);
+		result = new ListItr(state=Created,
+		parent = self,
+		index = self.index);
+	}
+
+
+	fun clone (): Object
+    	{ 
+		result = action LIST_DUP(storage);
+	}
+	
+	
+	fun hashCode (): int
+    	{
+       	 	// result = action OBJECT_HASH_CODE(self);
+       		// #problem
+       		action NOT_IMPLEMENTED();
+    	}
+	
+	
+	fun toString (): string
+  	{
+       		// result = action OBJECT_TO_STRING(self);
+       		// #problem
+        	action NOT_IMPLEMENTED();
+    	}
+
+}
+
+
+
+
+
+
+@Private
+@Implements('java.util.ListIterator')
+@WrapperMeta(
+    src='java.util.ListItr',
+    dst='org.utbot.engine.overrides.collections.UtListItr',
+    matchInterfaces=true,
+)
+automaton ListItr: int(
+	var nextIndex:int,
+	var next: Object,
+	var lastReturned: Object,
+	var expectedModCount = parent.modCount)
+{
+	initstate Initialized;
+	
+	shift Initialized -> self by [
+        // read operations
+        hasNext,
+	hasPrevious,
+	nextIndex,
+	previousIndex,
+	
+        // write operations
+        next,
+        remove,
+	previous,
+	set,
+	add,
+	forEachRemaining
+    ];
+    
+    //constructors
+    
+    constructor ListItr(index: int): void 
+    {
+    	if(index == self.parent.size)
```

Comment:
But what will happen if `index` is greater than `size`? (a hint)

Reply:
Has been fixed.